### DISCUSSION TITLE
Title of HTML Document

### DISCUSSION THREAD

**Discussion by ghost** [mecha-cms/mecha#121] [Q&A]
Hai Mas Taufik,

Selamat berlebaran dan mohon maaf lahir batin 🙏.

Saya mau tanya tentang dokumentasi kode `<?= w($t->reverse); ?>`. Di dan dari manakah saya bisa memulai memodifikasi judul dokumen HTML? Misalnya saya hanya ingin menampilkan judul postingan tanpa embel-embel nama situs?

Atau dalam kasus: jika judul postingan di YAML memiliki jumlah karakter lebih dari 60 karakter, maka tampilkan judul postingan tanpa embel-embel nama situs di judul dokumen HTML. Jika judul postingan di YAML jumlahnya kurang dari 60 karakter, maka judul dokumen HTML tampilkan nama situs.

Kalau di Blogger saya biasa menggunakan `.length`.

**Comment by taufik-nurrohman** [comment]
> Selamat berlebaran dan mohon maaf lahir batin.

_Taqabbalallahu minna wa minkum._ Semoga Allah menerima puasa kita dan setiap tahun semoga kita senantiasa dalam kebaikan.

> Saya mau tanya tentang dokumentasi kode `<?= w($t->reverse); ?>`

Variabel `$t` sebenarnya cuma _array_ yang diterapkan ke kelas `Anemon`. Jadi kalau mas baca dokumentasi soal kelas ini, semua metode di dalamnya sebenarnya bisa dipakai. Di sini Saya pakai `reverse()` supaya judul artikel muncul lebih dulu dari judul situs web. Fungsi `w()` itu cuma sebagai _guard_ supaya elemen-elemen HTML di dalam data `$t` bisa dihapus karena kita pakai itu di dalam elemen `<title>`.

> Misalnya saya hanya ingin menampilkan judul postingan tanpa embel-embel nama situs?

Pakai ini:

~~~ php
<title>
  <?= w($page->title ?? $site->title); ?>
</title>
~~~

> Jika judul postingan di YAML memiliki jumlah karakter lebih dari 60 karakter, maka tampilkan judul postingan tanpa embel-embel nama situs di judul dokumen HTML. Jika judul postingan di YAML jumlahnya kurang dari 60 karakter, maka judul dokumen HTML tampilkan nama situs.

~~~ php
<?php if (gt((string) $page->title, 60)): ?>
  <title>
    <?= w($page->title); ?>
  </title>
<?php else: ?>
  <title>
    <?= w($t->reverse); ?>
  </title>
<?php endif; ?>
~~~
